tests/test_flow.py

```diff
@@ -5015,3 +5015,92 @@ def __init__(self, executor: RunnableExecutor):
         "busy2": 1,
         "thread1": 1,
     }
+
+
+def test_enrichment():
+    busy_wait_pool = RunnableBusyWait("busy1")
+    busy_wait_dedicated = RunnableBusyWait("busy2")
+    busy_wait_dedicated.execution_mechanism = "dedicated_process"
+
+    runnables = [
+        busy_wait_pool,
+        busy_wait_dedicated,
+    ]
+
+    class MyParallelExecution(ParallelExecution):
+
+        def enrich_event(self, event):
+            event._metadata = {"name": self.name}
+            return event
+
+        def select_runnables(self, event):
+            return [runnable.name for runnable in runnables if runnable.name != "error"]
+
+    parallel_execution = MyParallelExecution(runnables, full_event=True)
+    reduce = Reduce([], lambda acc, x: acc + [x], full_event=True)
+
+    source = SyncEmitSource()
+    source.to(parallel_execution).to(reduce)
+
+    start = time.monotonic()
+    controller = source.run()
+    controller.emit(0)
+    controller.terminate()
+    termination_result = controller.await_termination()
+    end = time.monotonic()
+
+    assert end - start < 3
+    result = termination_result[0].body
+    total_metadata = termination_result[0]._metadata
+
+    assert result == {
+        "busy1": 1,
+        "busy2": 1,
+    }
+    assert (
+        "name" in total_metadata and total_metadata.pop("name") == "MyParallelExecution"
+    ), "Expected name in _metadata field"
+    assert all(
+        list(("when" in metadata and "microsec" in metadata) for metadata in total_metadata.values())
+    ), "Expected _metadata to include 'when' and 'microsec' fields "
+
+
+def test_without_enrichment():
+    busy_wait_pool = RunnableBusyWait("busy1")
+    busy_wait_dedicated = RunnableBusyWait("busy2")
+    busy_wait_dedicated.execution_mechanism = "dedicated_process"
+
+    runnables = [
+        busy_wait_pool,
+        busy_wait_dedicated,
+    ]
+
+    class MyParallelExecution(ParallelExecution):
+
+        def select_runnables(self, event):
+            return [runnable.name for runnable in runnables if runnable.name != "error"]
+
+    parallel_execution = MyParallelExecution(runnables, full_event=True)
```

Comment:
I don't think full_event is observed by ParallelExecution. It always receives the full event and hands off just the body and path to the runnables.